enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/ColorCustomizationSection.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback, useState } from "react";
+import { t } from "ttag";
+
+import { ColorPillPicker } from "metabase/common/components/ColorPicker";
+import { useSetting } from "metabase/common/hooks";
+import type { MetabaseColors } from "metabase/embedding-sdk/theme";
+import { colors as defaultMetabaseColors } from "metabase/lib/colors";
+import { ActionIcon, Group, Icon, Stack, Text, Tooltip } from "metabase/ui";
+
+import { getConfigurableThemeColors } from "../utils/theme-colors";
+
+interface ColorCustomizationSectionProps {
+  theme?: { colors?: Partial<MetabaseColors> };
+  onColorChange: (colors: Partial<MetabaseColors>) => void;
+  onColorReset?: () => void;
+}
+
+export const ColorCustomizationSection = ({
+  theme,
+  onColorChange,
+  onColorReset,
+}: ColorCustomizationSectionProps) => {
+  const applicationColors = useSetting("application-colors");
+
+  // Undebounced color values to keep color selection fast.
+  const [colorPreviewValues, setColorPreviewValues] = useState<
+    Record<string, string>
+  >({});
```

Comment:
We need to lift the preview colors from `ColorPillPicker` here so we can reset it. I could have used the key trick but this is cleaner.